resolwe/storage/connectors/s3connector.py

```diff
@@ -364,3 +371,59 @@ def presigned_url(self, url, expiration=60, force_download=False):
 
         # The response contains the presigned URL
         return response
+
+    def temporary_credentials(
+        self, prefix: str, duration: int = 900, role_arn: Optional[str] = None, **kwargs
+    ) -> Dict:
+        """Get the temporary credentials for role_arn.
+
+        The credentials are further limited to access only objects which names
+        start with '/prefix/'.
+
+        :attr role_arn: arn of the role to assume. If none is given it is taken
+            from connector settings.
+
+        :attr prefix: the string representing prefix.
+
+        :attr duration: the duration of the credentials in seconds (defaults to 60).
+
+        :raises KeyError: when no role_arn is gives as argument or connector
+            setting.
+
+        :returns: the dictionary with credentials, see
+        https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts.html#STS.Client.assume_role
+        for detailed description.
+        """
+        # Policy to add access only to the objects named '/prefix/*'.
+        policy = (
+            """{
+ "Version":"2012-10-17",
+ "Statement": [
+   {
+     "Sid": "AllowAllS3ActionsInUserFolder",
+     "Effect": "Allow",
+     "Action": ["s3:*"],
+     "Resource": ["arn:aws:s3:::"""
+            + self.bucket_name
+            + "/"
+            + prefix
+            + """/*"]
+   },
+   {
+     "Sid": "AllowUserKMSUsage",
+     "Effect": "Allow",
+     "Action": ["kms:*"],
+     "Resource": ["*"]
+   }
+]
+}"""
+        )
+        result = self.original_session.client("sts").assume_role(
+            RoleArn=role_arn or self.config["role_arn"],
+            RoleSessionName="upload-" + prefix,
+            DurationSeconds=duration,
+            Policy=policy,
+        )
+        result["bucket_name"] = self.bucket_name
+        result["region"] = self.config.get("region_name")
+        return result
```

Comment:
How will the client know credentials for which service are this? It would be nice to also include a service identifier and any other data the client will need to perform the upload.

Reply:
From the endpoint, which is meant exclusively for upload credentials and the fact we are using only AWS S3 buckets.

Were you thinking about adding some identifier so it would be clear the credentials are for AWS S3 service?